src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -121,16 +122,136 @@ public Optional<String> getAssignment(
      * @param experimentKey
      * @return
      */
-    public Optional<String> getAssignment(String subjectKey, String experimentKey) {
+    private Optional<EppoValue> getAssignment(String subjectKey, String experimentKey) {
         return this.getAssignment(subjectKey, experimentKey, new SubjectAttributes());
     }
 
+    /**
+     * This function will return typed assignment value
+     * @param subjectKey
+     * @param experimentKey
+     * @param type
+     * @param subjectAttributes
+     * @return
+     */
+    private Optional<?> getTypedAssignment(String subjectKey, String experimentKey, EppoValueType type, SubjectAttributes subjectAttributes) {
+        Optional<EppoValue> value = this.getAssignment(subjectKey, experimentKey, subjectAttributes);
+        if (value.isEmpty()) {
+            return Optional.empty();
+        }
+
+        switch (type) {
+            case BOOLEAN:
+                return Optional.of(value.get().boolValue());
+            case NUMBER:
+                return Optional.of(value.get().longValue());
```

Comment:
Do we not have to worry about numbers with decimals? Looking at the existing code, it seems we don't worry about them, but I figure I'd flag it.

Reply:
@aarsilv Thanks for flagging this. I will change it to decimal.